cocos/gfx/device-manager.ts

```diff
@@ -155,13 +157,15 @@ export class DeviceManager {
             return;
         }
 
+        if (this._canvas) { this._canvas.oncontextmenu = () => false; }
+    }
+
+    private _initSwapchain () {
         const swapchainInfo = new SwapchainInfo(this._canvas!);
         const windowSize = screen.windowSize;
         swapchainInfo.width = windowSize.width;
         swapchainInfo.height = windowSize.height;
         this._swapchain = this._gfxDevice.createSwapchain(swapchainInfo);
-
-        if (this._canvas) { this._canvas.oncontextmenu = () => false; }
```

Comment:
Why remove it?

Reply:
It is in the final line of `DeviceManager.init()`.